### PR TITLE
Fix WeekOfMonth constructor

### LINKED ISSUE #629: WeekOfMonth typing is incorrect
Typing should be what is in `pandas`:
```python
class WeekOfMonth(WeekOfMonthMixin):
    def __init__(
        self, n: int = ..., normalize: bool = ..., week: int = ..., weekday: int = ...
    ) -> None: ...
```
Currently looks like:
https://github.com/pandas-dev/pandas-stubs/blob/9c301b33b746e28401868b788364cecd00a34655/pandas-stubs/_libs/tslibs/offsets.pyi#L182

### PR DESCRIPTION
**Pull request by twoertwein**
- [x] Closes #629 (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: tests/test_timefuncs.py [hunk 1]
```diff
@@ -635,7 +635,7 @@ def test_some_offsets() -> None:
         pd.DatetimeIndex,
     )
     # GH 320
-    tswm1 = pd.Timestamp("9/23/2022") + pd.offsets.WeekOfMonth(2, 3)
+    tswm1 = pd.Timestamp("9/23/2022") + pd.offsets.WeekOfMonth(2, True)
```
Comment (twoertwein) [inline tests/test_timefuncs.py:638]: Pandas probably treated 3 as True.

File: tests/test_timefuncs.py [hunk 2]
```diff
@@ -1141,3 +1141,8 @@ def test_timestamp_strptime_fails():
                 "%Y-%M-%D",  # type: ignore[arg-type] # pyright: ignore[reportGeneralTypeIssues]
             )
         )
+
+
+def test_weekofmonth_init():
+    # GH 629
+    pd.offsets.WeekOfMonth(n=1, week=1, weekday=1, normalize=True)
```
Comment (twoertwein) [inline tests/test_timefuncs.py:1148]: Does this need a check/assert_type statement?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Add `check(assert_type(...`

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein

**Pull request merged**